### ISSUE TITLE
Observabillity: Investigate why our spans are so big

### ISSUE THREAD

**Issue by burmudar** [sourcegraph/sourcegraph#43959]
Granfana Cloud and Honeycomb either discard or complain about our spans being too big. See below for a dashboard showcasing spans being discarded on Grafana Cloud.

![Screenshot 2022-11-04 at 17 59 57](https://user-images.githubusercontent.com/1001709/200021441-31cac79e-3285-4e80-959e-c3ab30c0d9af.png)

Even though we're moving away from Grafana Cloud, this is in issue we will surely hit with other providers and we need to understand why it's happening and address it.

**Comment by burmudar** [comment]
Additional things to look into
<img width="1221" alt="CleanShot 2022-11-08 at 13 29 30@2x" src="https://user-images.githubusercontent.com/1001709/200843480-c4667575-e715-4f87-b032-bf329ece279e.png">

**Comment by burmudar** [comment]
In the following PR `sql.Rows` spans were turned off, which resulted in less useful spans being sent
- https://github.com/sourcegraph/sourcegraph/pull/44798

This helped in reducing the amount of spans across the board.

Another cause for our large traces/spans is due to two factors
1. Distributed systems
2. Streaming

Due to the nature of distributed systems a trace can grow by N where N is the amount of distributed systems that are contacted. For instance - when a query goes out to zoekt, all zoekt instances received the query and process it. If one zoekt instances produces 1k spans, it means the overall trace will grow by how many zoekt instances we have in the cluster ie. 8 zoekt shards means 8k spans get added per query. And the fact that results are streamed, these queries don't just start and end, but can go on for 30s and up. 

Thus assuming, 1 zoekt instance generates 50 spans for 1 resultset, and in a stream of 30 secs, 1 resultset is returned every 1s we get: `50 spans * 30s  = 1500 spans`. Now if we add that we have 10 zoekt instances, each doing the same thing `1500 spans * 10 zoekts = 15000` we can easily get traces that have 15k spans. This is aligns with what I have observed from the debug endpoint on the collector `/debug/tracez`.
```
2022/11/24-10:21:33.746980    10.333027 trace_id: 4822ecc1ca6420b61609c9b2034ce20a span_id: f98a8521b30a9af1
                                        Status{Code=Unset, description=""}
                                        Attributes:{send_failed_spans=0, sent_spans=46168}
```

TL;DR: Some of our traces are large because they go to lots of systems and with streaming they produce traces constantly and don't end with one request - meaning the trace grows over a period instead of staying constant per request. Thus reducing spans that get added to a trace (as was done in the PR) can have a big consequence.
